### ISSUE TITLE
Steam wiped all my applications

### ISSUE THREAD

**Issue by igorbispo99** [ValveSoftware/steam-for-linux#6617]
* Steam client version (build number or date): I dont know (l just downloaded the last release from https://store.steampowered.com/about)
* Distribution (e.g. Ubuntu): Debian 10 Buster
* Opted into Steam client beta?: No
* Have you checked for system updates?: [Yes/No] Yes

So, i downloaded the "steam_latest.deb" file from https://store.steampowered.com/about/, installed it normally using "sudo dpkg -i steam_latest.deb" .
Then, when i ran the app using "steam", an xterm window popped up asking for the sudo password to install some libraries (which is very common and has happened to me several times during other installations of Steam in another distros). 
After that, a second xterm window appeared, now, asking for me to type a kind of confirmation text (something like "I agree, you can do it"). Then, apt started to remove some applications ( i first tought it was normal until it removed things like g++, chrome, drivers, desktop environment and the kernel itself) i end up hiting CTRL+C to interupt the process but it was to late and the SO dont boot up anymore.

#### Steps for reproducing this issue:

1. Install Steam for Linux
2. Run it for the first time
3. Insert sudo password to install libraries
4. Insert the confirmation text

**Comment by baryluk** [comment]
That is bad.

Valve, I assume steam is executing `apt-get --assume-yes install xyz:i386 ...` , and due some conflicting versions it decided to uninstall some other packages to full fill dependencies.

How about also adding `--no-remove`` to prevent breakage of the unrelated apps?

**Comment by theokktok** [comment]
@baryluk The same way it was “kind of my fault” it was “kind of software’s fault” since that kind of destructive operation should never happen during a regular Steam installation. To be honest, I’ve reported this issue to prevent future damage to another people, I’m not saying that it was anyone’s fault or Valve/devs should be punished or anything like that.

**Comment by baryluk** [comment]
> @baryluk The same way it was “kind of my fault” it was “kind of software’s fault” since that kind of destructive operation should never happen during a regular Steam installation. To be honest, I’ve reported this issue to prevent future damage to another people, I’m not saying that it was anyone’s fault or Valve/devs should be punished or anything like that.

That is why I suggested adding `--no-remove` to apt-get invocation in my other comment.

**Comment by kisak-valve** [comment]
Hello @igorbispo99, are you still experiencing this issue on an up to date system?

The 1.0.0.72 Steam client package includes this in its changelog:
```
  * bin_steamdeps: Ensure that packages are never removed.
    In some situations, e.g. if a user installs a newer version of
    libc6:amd64 and then removes its apt source, resulting in it being
    newer than any available version of libc6:i386, apt would suggest
    removing several essential packages in order to satisfy the
    dependencies. Don't let it do that.
    (Resolves: steam-for-linux#7982)
  * bin_steamdeps: Explicitly ask apt for both :amd64 and :i386 versions of
    some required packages, so that it will give better error messages in
    situations like the one that promoted steam-for-linux#7982.
```

**Comment by kisak-valve** [comment]
Closing as fixed in in the Valve provided Steam package. (Note: Older distro-provided variants would not have been updated.)
